pkg/lexer/lexer.go

```diff
@@ -0,0 +1,84 @@
+package lexer
```

Comment:
This is my lexer - let me know what you think. I'm using the functional (somewhat) design to keep the loop as clean as possible.